exercises/diamond/uDiamondTest.pas

```diff
@@ -0,0 +1,149 @@
+unit uDiamondTest;
+
+interface
+uses
+  DUnitX.TestFramework;
+
+const
+  CanonicalVersion = '1.1.0';
+
+type
+  [TestFixture]
+  TDiamondTest = class(TObject)
+  private
+    Expected : TArray<string>;
+    procedure CompareArrays(Array1, Array2: TArray<string>);
+  public
+    [Test]
+//    [Ignore('Comment the "[Ignore]" statement to run the test')]
+    procedure Degenerate_case_with_a_single_A_row;
+
+    [Test]
+    [Ignore]
+    procedure Degenerate_case_with_no_row_containing_3_distinct_groups_of_spaces;
+
+    [Test]
+    [Ignore]
+    procedure Smallest_non_degenerate_case_with_odd_diamond_side_length;
+
+    [Test]
+    [Ignore]
+    procedure Smallest_non_degenerate_case_with_even_diamond_side_length;
+
+    [Test]
+    [Ignore]
+    procedure Largest_possible_diamond;
+  end;
+
+implementation
+
+uses
+  System.SysUtils, uDiamond;
+
+{ TDiamondTest }
+
+procedure TDiamondTest.CompareArrays(Array1, Array2: TArray<string>);
+var
+  i: integer;
+begin
+  Assert.AreEqual(Length(Array1), Length(Array2), ' - Array lengths must be equal');
+  for i := Low(Array1) to High(Array1) do
+    Assert.AreEqual(Array1[i], Array2[i], format('Expecting element %d to = %s, Actual = %s',
+      [i, Array1[i], Array2[i]]));
+end;
+
+procedure TDiamondTest.Degenerate_case_with_a_single_A_row;
+begin
+  Expected := ['A'];
+  CompareArrays(Expected, TDiamond.Rows('A'));
+end;
+
+procedure TDiamondTest.Degenerate_case_with_no_row_containing_3_distinct_groups_of_spaces;
+begin
+  Expected := [ ' A ',
```

Comment:
Very simple white space correction.  Space between `[` and `'` should be removed.